File: server/src/main/scala/producer/message_generator/ordering_key_generator/OrderingKeyGenerator.scala
package producer.message_generator.ordering_key_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb
import producer.message_generator.data_generators.bytes.BytesGenerator

case class OrderingKeyGenerator(
    fromBytes: BytesGenerator
)

object OrderingKeyGenerator:
    def fromPb(v: pb.OrderingKeyGenerator): OrderingKeyGenerator =
        OrderingKeyGenerator(
            fromBytes = BytesGenerator.fromPb(v.fromBytes.get)
        )

    def toPb(v: OrderingKeyGenerator): pb.OrderingKeyGenerator =
        pb.OrderingKeyGenerator(
            fromBytes = Some(BytesGenerator.toPb(v.fromBytes))
        )

File: ui/components/ui/ProducerSession/producer-task/message-generator/ordering-key-generator/ordering-key-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';
import { BytesGenerator, bytesGeneratorFromPb, bytesGeneratorToPb } from '../data-generators/bytes/bytes-generator';

export type OrderingKeyGenerator = {
  type: 'ordering-key-generator',
  fromBytes: BytesGenerator
};

export function orderingKeyGeneratorFromPb(v: pb.OrderingKeyGenerator): OrderingKeyGenerator {
  return {
    type: 'ordering-key-generator',
    fromBytes: bytesGeneratorFromPb(v.getFromBytes()!)
  };
}

export function orderingKeyGeneratorToPb(v: OrderingKeyGenerator): pb.OrderingKeyGenerator {
  const generatorPb = new pb.OrderingKeyGenerator();
  generatorPb.setFromBytes(bytesGeneratorToPb(v.fromBytes));

  return generatorPb;
}
